src/main/java/core/basesyntax/service/impl/FruitRecordParserImpl.java

```diff
@@ -0,0 +1,53 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.dto.FruitRecordDto;
+import core.basesyntax.service.FruitRecordParser;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FruitRecordParserImpl implements FruitRecordParser {
+    private static final int OPERATION_TYPE = 0;
+    private static final int FRUIT_NAME = 1;
+    private static final int QUANTITY = 2;
+    private static final int CORRECT_FIELDS = 3;
+    private static final String SPLITTER = ",";
+    private static final String ERROR = "Data is not correct";
+
+    @Override
+    public List<FruitRecordDto> parse(List<String> strings) {
```

Comment:
You should check data for validity in this method.
for example - if you get negative number in amount.
or if you get an unvalid letter for operation type

Reply:
done